plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -753,10 +753,14 @@ private ConnectorTableMetadata doGetTableMetadata(ConnectorSession session, Sche
             }
             catch (TrinoException e) {
                 // Skip this table if there's a failure due to Hive, a bad Serde, or bad metadata
-                if (!e.getErrorCode().getType().equals(ErrorType.EXTERNAL)) {
-                    throw e;
+                if (e.getErrorCode().getType().equals(ErrorType.EXTERNAL)) {
```

Comment:
You are changing here the original behavior to fit in the handling for unsupported table types.

What about adding an `UnsupportedTableTypeException` (with `UNSUPPORTED_TABLE_TYPE` error code) to `trino-spi` and replace:

- NotADeltaLakeTableException (trino-delta-lake)
- UnknownTableTypeException (trino-iceberg)

This would probably keep the original error flow of this method almost untouched (special handling for `UnsupportedTableTypeException`)